### PR TITLE
Checking in autonat documentation for review

### PR DESCRIPTION
**Pull request by efrisch**
## Description

Fixes pyrsia/pyrsia#943

This PR documents our bootstrap process and the libp2p autonat negotiation for developers.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

N/A

### CODE REVIEW

File: docs/developers/pyrsia-autonat.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Pyrsia AutoNAT Implementation (ietf STUN)
+
+> **Warning:** This document is a work-in-progress.
+
+## Abbreviations used in this document
+
+- NAT: Network Address Translation. A mechanism for connecting and routing between private networks and the public internet.
+- STUN: Session Traversal Utilities for NAT. An ietf standard for connecting applications over the public internet originating from private networks.
```
Comment (karolh2000) [inline docs/developers/pyrsia-autonat.md:8]: minor - change ietf to IETF

File: docs/developers/pyrsia-autonat.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- IETF: Internet Engineering Taskforce. One of the groups that standardizes protocols on the Internet.
```
Comment (karolh2000) [inline docs/developers/pyrsia-autonat.md:9]: minor - Taskforce vs Task Force.

**Comment by sbtaylor15** [comment]
@efrisch Is it possible to use a mermaid markdown instead of the png?  That way we can change it easily down the road if needed.

**Comment by erwin1** [comment]
@efrisch this could be as simple as this (although not 100% the same)

```mermaid
graph TD;
node[Pyrsia Node] --HTTP GET /status--> boot[boot.pyrsia.link]
boot --> boot1[Boot node 1<br>PeerId A]
boot --> boot2[boot node 2<br>PeerId B]
boot --> boot3[Boot node 3<br>PeerId C]
boot --> boot4[Boot node 4<br>PeerId D]
boot1 --> json[JSON<br>Containing peer<br>addresses and<br>basic metrics]
boot2 --> json
boot3 --> json
boot4 --> json
json ----> node
```

**Comment by erwin1** [comment]
@efrisch I read what you have so far and that's looking good. What I would maybe like to read in such documentation is how things work after bootstrapping. Maybe also with a little (mermaid) diagram.

**Review by erwin1 — APPROVED** [review]

**Pull request merged**